### PR TITLE
Completed T9F2-18

### PR DESCRIPTION
**Pull request by Romangino**
Created GroupDao and updated GroupDaoI to incorporate changes made.

### CODE REVIEW

**Comment by mitchneides** [comment]
Looks good to me!

File: daos/GroupDao.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    private constructor() {
+    }
+
+    /**
+     * Removes group object with given primary key from database.
+     * @param {string} uid Primary key of user
+     * @param {string} gid Primary key of group
+     * @returns Promise To be notified when group is removed from the database
+     */
+    deleteGroup = async (uid: string, gid: string): Promise<any> =>
```
Comment (mitchneides) [inline daos/GroupDao.ts:37]: In rethinking this - do we need a user id here?
Couldn't we just take the session's active user id to accomplish the same?

File: daos/GroupDao.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        GroupModel.deleteOne({_id: gid})
+
+    /**
+     * Finds all groups with both active user and other user
+     * @param {string} active_uid Primary key of active user
+     * @param {string} other_uid Primary key of other user
+     * @returns Promise To be notified when group is retrieved from database
+     */
+    findAllCommonGroupsAnotherUser = async (active_uid: string, other_uid: string): Promise<Group[]> =>
```
Comment (mitchneides) [inline daos/GroupDao.ts:46]: Same with here - do we need to pass active uid or can we just get from the session info?

File: daos/GroupDao.ts (hunk 1 above)
Comment (Romangino) [reply daos/GroupDao.ts:37]: Yes we can probably just use gid to delete group and verify user in session is admin

File: daos/GroupDao.ts (hunk 2 above)
Comment (Romangino) [reply daos/GroupDao.ts:46]: We will need to check Group object if both user in session and other id is in group so i think we might need it

**Comment by mitchneides** [comment]
Made a few comments but I think we need to discuss them because from the backend standpoint I think we don't need to pass the active user id in the API calls since we can just grab that info from the session info (I think but maybe someone else knows for sure?).
However, from a postman standpoint it may be difficult to test without passing that info in the URL.
What do you guys think?

File: daos/GroupDao.ts (hunk 2 above)
Comment (mitchneides) [reply daos/GroupDao.ts:46]: I meant just for the active user id cause I think we can get that from the session info but agree we'll definitely still need the other id.
Made a post below about testing difficulties without passing both though so let me know what you think

Comment (Romangino) [reply daos/GroupDao.ts:46]: I tested GroupModel.find({members: {$all: [active_uid, other_uid]}}) in mongo shell with some mock groups and was able to find all groups with both users in it. I think just having the other_uid will give us all groups other_uid is in and not the current user. We could use the session info to pass in the current user's id and return mutual groups

File: daos/GroupDao.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    findGroupByName = async (group_name: string, uid: string): Promise<Group[]> =>
+        GroupModel.find({groupName: group_name, members: uid})
+
+    /**
+     * Finds group by group id
+     * @param {string} gid Primary key of group
+     * @returns Promise To be notified when group is retrieved
+     */
+    findGroupByGroupId = async (gid: string): Promise<any> =>
```
Comment (mitchneides) [inline daos/GroupDao.ts:71]: Shouldn't this promise be <Group> ?

**Comment by adityaputtigampala** [comment]
ideally it would be nice to grab the user info from the logged in session, but yea to the point about testing I'm not sure how we would have it integrated. What do you think about adding it to the backlog as a potential code refactor if we have time, but moving forward with the current implementation for now. We could potentially leverage this code

https://www.back4app.com/docs/react/working-with-users/get-current-user-react

**Comment by Romangino** [comment]
Is this branch ok to merge?

**Comment by spencerbelsky** [comment]
Not sure if the conversation above about the user info was resolved, but it sounds like we want to keep it in for now and if we decided it is unnecessary we can remove it?

**Comment by mitchneides** [comment]
Yeah agreed I think we're all on the same page, I vote go ahead and merge and then I'll PR the controller so we can all do our tests and then hopefully merge that too and finish this sprint

**Pull request merged**